File: bindings/smartvaults-sdk-ffi/bindings-python/examples/restore.py
from smartvaults_sdk import SmartVaults, Network, SyncHandler, init_desktop_logger
import time

init_desktop_logger("/home/user/.smartvaults", Network.TESTNET)

client = SmartVaults.restore("/home/user/.smartvaults", "name", "test", "mnemonic", None, Network.TESTNET)

class SyncNotifications(SyncHandler):
    def handle(self, msg):
        print("Refreshing...")
        vaults = client.vaults()
        for p in vaults:
            print(p.policy().name())

handle = client.handle_sync(SyncNotifications())

while True:
    time.sleep(10.0)

File: crates/smartvaults-sdk/examples/restore.rs
// Copyright (c) 2022-2024 Smart Vaults
// Distributed under the MIT software license

use std::str::FromStr;

use smartvaults_sdk::prelude::*;

const NETWORK: Network = Network::Testnet;

#[tokio::main]
async fn main() {
    // Restore and initialize a new Smart Vaults client
    let client = SmartVaults::restore(
        "./your-path",
        "account-name",
        || Ok(String::from("password")),
        || Ok(String::from("confirm-password")),
        || Ok(Mnemonic::from_str("your menmonic").unwrap()),
        || Ok(None),
        NETWORK,
    )
    .await
    .unwrap();

    // Check the others examples to learn more about the client APIs

    // Shutdown the client (for logout)
    client.shutdown().await.unwrap();
}
